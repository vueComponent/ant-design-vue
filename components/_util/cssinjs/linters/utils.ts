import devWarning from '../../../vc-util/warning';
import type { LinterInfo } from './interface';

export function lintWarning(message: string, info: LinterInfo) {
  const { path, parentSelectors } = info;

  devWarning(
    false,
    `[Ant Design Vue CSS-in-JS] ${path ? `Error in '${path}': ` : ''}${message}${
      parentSelectors.length ? ` Selector info: ${parentSelectors.join(' -> ')}` : ''
    }`,
  );
}
