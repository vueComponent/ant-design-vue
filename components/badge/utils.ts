import { PresetColorTypes, PresetStatusColorTypes } from '../_util/colors';

export function isPresetColor(color?: string): boolean {
  return (
    (PresetColorTypes as any[]).indexOf(color) !== -1 ||
    (PresetStatusColorTypes as any[]).indexOf(color) !== -1
  );
}
