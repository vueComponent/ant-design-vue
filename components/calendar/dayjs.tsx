import generateConfig from '../vc-picker/generate/dayjs';
import type { CalendarProps } from './generateCalendar';
import generateCalendar from './generateCalendar';

const Calendar = generateCalendar(generateConfig);

export type { CalendarProps };
export default Calendar;
