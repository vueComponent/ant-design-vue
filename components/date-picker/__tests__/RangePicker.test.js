import { mount } from '@vue/test-utils';
import { asyncExpect, setMockDate, resetMockDate, sleep } from '../../../tests/utils';
import dayjs from 'dayjs';
import DatePicker from '../';
import { openPicker, selectCell, closePicker } from './utils';
import focusTest from '../../../tests/shared/focusTest';
jest.mock('../../_util/Portal');
const { RangePicker } = DatePicker;

describe('RangePicker', () => {
  focusTest(RangePicker);

  beforeEach(() => {
    document.body.outerHTML = '';
    setMockDate();
  });

  afterEach(() => {
    resetMockDate();
  });
  // issue: https://github.com/ant-design/ant-design/issues/5872
  xit('should not throw error when value is reset to `[]`', async () => {
    const birthday = dayjs('2000-01-01', 'YYYY-MM-DD');
    const wrapper = mount(
      {
        props: ['value'],
        setup(props) {
          return () => <RangePicker value={props.value} open />;
        },
      },
      { attachTo: 'body', sync: 'false' },
    );
    wrapper.setProps({ value: [birthday, birthday] });
    wrapper.setProps({ value: [] });
    await sleep(0);
    expect(() => {
      openPicker(wrapper);
      selectCell(wrapper, 3);
      closePicker(wrapper);

      openPicker(wrapper, 1);
      selectCell(wrapper, 5, 1);
      closePicker(wrapper, 1);
    }).not.toThrow();
  });

  it('show month panel according to value', async () => {
    const birthday = dayjs('2000-01-01', 'YYYY-MM-DD').locale('zh-cn');
    const wrapper = mount(RangePicker, {
      props: {
        getPopupContainer: trigger => trigger,
        format: 'YYYY/MM/DD',
        showTime: true,
        open: true,
      },
      sync: false,
      attachTo: 'body',
    });
    await asyncExpect(() => {
      wrapper.setProps({ value: [birthday, birthday] });
    });
    await asyncExpect(() => {
      expect(document.body.innerHTML).toMatchSnapshot();
    });
  });

  fit('customize separator', async () => {
    const wrapper = mount(RangePicker, { props: { separator: 'test' } });
    await sleep();
    expect(wrapper.html()).toMatchSnapshot();
  });

  fit('test WeekPicker valueFormat', async () => {
    const case1 = ['2023-22', '2023-24'];
    const case2 = ['2023-27', '2023-28'];
    const wrapper = mount(RangePicker, {
      props: {
        picker: 'week',
        format: 'YYYY-ww',
        valueFormat: 'YYYY-ww',
        value: case1,
      },
      sync: false,
      attachTo: 'body',
    });
    const inputs = wrapper.findAll('input');
    expect((inputs[0].element.value, inputs[1].element.value)).toBe((case1[0], case1[1]));
    await asyncExpect(() => {
      wrapper.setProps({ value: case2 });
    });
    expect((inputs[0].element.value, inputs[1].element.value)).toBe((case2[0], case2[1]));
  });
});
