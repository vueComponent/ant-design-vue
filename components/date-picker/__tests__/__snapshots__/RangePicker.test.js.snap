// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangePicker customize separator 1`] = `
<div class="ant-picker ant-picker-range">
  <div class="ant-picker-input ant-picker-input-active"><input readonly="" placeholder="Start date" size="12" autocomplete="off"></div>
  <div class="ant-picker-range-separator">test</div>
  <div class="ant-picker-input"><input readonly="" placeholder="End date" size="12" autocomplete="off"></div>
  <div class="ant-picker-active-bar" style="left: 0px; width: 0px; position: absolute;"></div><span class="ant-picker-suffix"><span role="img" aria-label="calendar" class="anticon anticon-calendar"><svg focusable="false" class="" data-icon="calendar" width="1em" height="1em" fill="currentColor" aria-hidden="true" viewBox="64 64 896 896"><path d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"></path></svg></span>
  <!----></span>
  <!---->
  <div>
    <!---->
    <div class="ant-picker-dropdown ant-picker-dropdown-range ant-slide-up-enter ant-slide-up-enter-prepare" style="pointer-events: none; display: none;">
      <div class="ant-picker-range-wrapper ant-picker-date-range-wrapper" style="min-width: 0px;">
        <div class="ant-picker-range-arrow" style="left: 0px;"></div>
        <div class="ant-picker-panel-container" style="margin-left: 0px;">
          <div class="ant-picker-panels">
            <div tabindex="0" class="ant-picker-panel ant-picker-panel-focused">
              <div class="ant-picker-date-panel">
                <div class="ant-picker-header"><button type="button" tabindex="-1" class="ant-picker-header-super-prev-btn">«</button><button type="button" tabindex="-1" class="ant-picker-header-prev-btn">‹</button>
                  <div class="ant-picker-header-view"><button type="button" tabindex="-1" class="ant-picker-month-btn">Sep</button><button type="button" tabindex="-1" class="ant-picker-year-btn">2017</button></div><button type="button" tabindex="-1" class="ant-picker-header-next-btn" style="visibility: hidden;">›</button><button type="button" tabindex="-1" class="ant-picker-header-super-next-btn" style="visibility: hidden;">»</button>
                </div>
                <div class="ant-picker-body">
                  <table class="ant-picker-content">
                    <thead>
                      <tr>
                        <th>Su</th>
                        <th>Mo</th>
                        <th>Tu</th>
                        <th>We</th>
                        <th>Th</th>
                        <th>Fr</th>
                        <th>Sa</th>
                      </tr>
                    </thead>
                    <tbody>
                      <tr>
                        <td title="2017-08-27" class="ant-picker-cell">
                          <div class="ant-picker-cell-inner">27</div>
                        </td>
                        <td title="2017-08-28" class="ant-picker-cell">
                          <div class="ant-picker-cell-inner">28</div>
                        </td>
                        <td title="2017-08-29" class="ant-picker-cell">
                          <div class="ant-picker-cell-inner">29</div>
                        </td>
                        <td title="2017-08-30" class="ant-picker-cell">
                          <div class="ant-picker-cell-inner">30</div>
                        </td>
                        <td title="2017-08-31" class="ant-picker-cell ant-picker-cell-end">
                          <div class="ant-picker-cell-inner">31</div>
                        </td>
                        <td title="2017-09-01" class="ant-picker-cell ant-picker-cell-start ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">1</div>
                        </td>
                        <td title="2017-09-02" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">2</div>
                        </td>
                      </tr>
                      <tr>
                        <td title="2017-09-03" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">3</div>
                        </td>
                        <td title="2017-09-04" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">4</div>
                        </td>
                        <td title="2017-09-05" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">5</div>
                        </td>
                        <td title="2017-09-06" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">6</div>
                        </td>
                        <td title="2017-09-07" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">7</div>
                        </td>
                        <td title="2017-09-08" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">8</div>
                        </td>
                        <td title="2017-09-09" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">9</div>
                        </td>
                      </tr>
                      <tr>
                        <td title="2017-09-10" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">10</div>
                        </td>
                        <td title="2017-09-11" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">11</div>
                        </td>
                        <td title="2017-09-12" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">12</div>
                        </td>
                        <td title="2017-09-13" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">13</div>
                        </td>
                        <td title="2017-09-14" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">14</div>
                        </td>
                        <td title="2017-09-15" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">15</div>
                        </td>
                        <td title="2017-09-16" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">16</div>
                        </td>
                      </tr>
                      <tr>
                        <td title="2017-09-17" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">17</div>
                        </td>
                        <td title="2017-09-18" class="ant-picker-cell ant-picker-cell-in-view ant-picker-cell-today">
                          <div class="ant-picker-cell-inner">18</div>
                        </td>
                        <td title="2017-09-19" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">19</div>
                        </td>
                        <td title="2017-09-20" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">20</div>
                        </td>
                        <td title="2017-09-21" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">21</div>
                        </td>
                        <td title="2017-09-22" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">22</div>
                        </td>
                        <td title="2017-09-23" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">23</div>
                        </td>
                      </tr>
                      <tr>
                        <td title="2017-09-24" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">24</div>
                        </td>
                        <td title="2017-09-25" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">25</div>
                        </td>
                        <td title="2017-09-26" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">26</div>
                        </td>
                        <td title="2017-09-27" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">27</div>
                        </td>
                        <td title="2017-09-28" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">28</div>
                        </td>
                        <td title="2017-09-29" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">29</div>
                        </td>
                        <td title="2017-09-30" class="ant-picker-cell ant-picker-cell-end ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">30</div>
                        </td>
                      </tr>
                      <tr>
                        <td title="2017-10-01" class="ant-picker-cell ant-picker-cell-start">
                          <div class="ant-picker-cell-inner">1</div>
                        </td>
                        <td title="2017-10-02" class="ant-picker-cell">
                          <div class="ant-picker-cell-inner">2</div>
                        </td>
                        <td title="2017-10-03" class="ant-picker-cell">
                          <div class="ant-picker-cell-inner">3</div>
                        </td>
                        <td title="2017-10-04" class="ant-picker-cell">
                          <div class="ant-picker-cell-inner">4</div>
                        </td>
                        <td title="2017-10-05" class="ant-picker-cell">
                          <div class="ant-picker-cell-inner">5</div>
                        </td>
                        <td title="2017-10-06" class="ant-picker-cell">
                          <div class="ant-picker-cell-inner">6</div>
                        </td>
                        <td title="2017-10-07" class="ant-picker-cell">
                          <div class="ant-picker-cell-inner">7</div>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
              <!---->
            </div>
            <div tabindex="0" class="ant-picker-panel ant-picker-panel-focused">
              <div class="ant-picker-date-panel">
                <div class="ant-picker-header"><button type="button" tabindex="-1" class="ant-picker-header-super-prev-btn" style="visibility: hidden;">«</button><button type="button" tabindex="-1" class="ant-picker-header-prev-btn" style="visibility: hidden;">‹</button>
                  <div class="ant-picker-header-view"><button type="button" tabindex="-1" class="ant-picker-month-btn">Oct</button><button type="button" tabindex="-1" class="ant-picker-year-btn">2017</button></div><button type="button" tabindex="-1" class="ant-picker-header-next-btn">›</button><button type="button" tabindex="-1" class="ant-picker-header-super-next-btn">»</button>
                </div>
                <div class="ant-picker-body">
                  <table class="ant-picker-content">
                    <thead>
                      <tr>
                        <th>Su</th>
                        <th>Mo</th>
                        <th>Tu</th>
                        <th>We</th>
                        <th>Th</th>
                        <th>Fr</th>
                        <th>Sa</th>
                      </tr>
                    </thead>
                    <tbody>
                      <tr>
                        <td title="2017-10-01" class="ant-picker-cell ant-picker-cell-start ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">1</div>
                        </td>
                        <td title="2017-10-02" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">2</div>
                        </td>
                        <td title="2017-10-03" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">3</div>
                        </td>
                        <td title="2017-10-04" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">4</div>
                        </td>
                        <td title="2017-10-05" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">5</div>
                        </td>
                        <td title="2017-10-06" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">6</div>
                        </td>
                        <td title="2017-10-07" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">7</div>
                        </td>
                      </tr>
                      <tr>
                        <td title="2017-10-08" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">8</div>
                        </td>
                        <td title="2017-10-09" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">9</div>
                        </td>
                        <td title="2017-10-10" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">10</div>
                        </td>
                        <td title="2017-10-11" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">11</div>
                        </td>
                        <td title="2017-10-12" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">12</div>
                        </td>
                        <td title="2017-10-13" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">13</div>
                        </td>
                        <td title="2017-10-14" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">14</div>
                        </td>
                      </tr>
                      <tr>
                        <td title="2017-10-15" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">15</div>
                        </td>
                        <td title="2017-10-16" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">16</div>
                        </td>
                        <td title="2017-10-17" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">17</div>
                        </td>
                        <td title="2017-10-18" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">18</div>
                        </td>
                        <td title="2017-10-19" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">19</div>
                        </td>
                        <td title="2017-10-20" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">20</div>
                        </td>
                        <td title="2017-10-21" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">21</div>
                        </td>
                      </tr>
                      <tr>
                        <td title="2017-10-22" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">22</div>
                        </td>
                        <td title="2017-10-23" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">23</div>
                        </td>
                        <td title="2017-10-24" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">24</div>
                        </td>
                        <td title="2017-10-25" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">25</div>
                        </td>
                        <td title="2017-10-26" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">26</div>
                        </td>
                        <td title="2017-10-27" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">27</div>
                        </td>
                        <td title="2017-10-28" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">28</div>
                        </td>
                      </tr>
                      <tr>
                        <td title="2017-10-29" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">29</div>
                        </td>
                        <td title="2017-10-30" class="ant-picker-cell ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">30</div>
                        </td>
                        <td title="2017-10-31" class="ant-picker-cell ant-picker-cell-end ant-picker-cell-in-view">
                          <div class="ant-picker-cell-inner">31</div>
                        </td>
                        <td title="2017-11-01" class="ant-picker-cell ant-picker-cell-start">
                          <div class="ant-picker-cell-inner">1</div>
                        </td>
                        <td title="2017-11-02" class="ant-picker-cell">
                          <div class="ant-picker-cell-inner">2</div>
                        </td>
                        <td title="2017-11-03" class="ant-picker-cell">
                          <div class="ant-picker-cell-inner">3</div>
                        </td>
                        <td title="2017-11-04" class="ant-picker-cell">
                          <div class="ant-picker-cell-inner">4</div>
                        </td>
                      </tr>
                      <tr>
                        <td title="2017-11-05" class="ant-picker-cell">
                          <div class="ant-picker-cell-inner">5</div>
                        </td>
                        <td title="2017-11-06" class="ant-picker-cell">
                          <div class="ant-picker-cell-inner">6</div>
                        </td>
                        <td title="2017-11-07" class="ant-picker-cell">
                          <div class="ant-picker-cell-inner">7</div>
                        </td>
                        <td title="2017-11-08" class="ant-picker-cell">
                          <div class="ant-picker-cell-inner">8</div>
                        </td>
                        <td title="2017-11-09" class="ant-picker-cell">
                          <div class="ant-picker-cell-inner">9</div>
                        </td>
                        <td title="2017-11-10" class="ant-picker-cell">
                          <div class="ant-picker-cell-inner">10</div>
                        </td>
                        <td title="2017-11-11" class="ant-picker-cell">
                          <div class="ant-picker-cell-inner">11</div>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
              <!---->
            </div>
          </div>
          <!---->
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`RangePicker show month panel according to value 1`] = `
<div data-v-app="">
  <div>
    <!---->
    <div class="ant-picker-dropdown ant-picker-dropdown-range" style="pointer-events: none; display: none;">
      <div class="ant-picker-range-wrapper ant-picker-date-range-wrapper" style="min-width: 0px;">
        <div class="ant-picker-range-arrow" style="left: 0px;"></div>
        <div class="ant-picker-panel-container" style="margin-left: 0px;">
          <div class="ant-picker-panels">
            <div tabindex="0" class="ant-picker-panel ant-picker-panel-has-range ant-picker-panel-focused">
              <div class="ant-picker-datetime-panel">
                <div class="ant-picker-date-panel">
                  <div class="ant-picker-header"><button type="button" tabindex="-1" class="ant-picker-header-super-prev-btn">«</button><button type="button" tabindex="-1" class="ant-picker-header-prev-btn">‹</button>
                    <div class="ant-picker-header-view"><button type="button" tabindex="-1" class="ant-picker-month-btn">Jan</button><button type="button" tabindex="-1" class="ant-picker-year-btn">2000</button></div><button type="button" tabindex="-1" class="ant-picker-header-next-btn">›</button><button type="button" tabindex="-1" class="ant-picker-header-super-next-btn">»</button>
                  </div>
                  <div class="ant-picker-body">
                    <table class="ant-picker-content">
                      <thead>
                        <tr>
                          <th>Su</th>
                          <th>Mo</th>
                          <th>Tu</th>
                          <th>We</th>
                          <th>Th</th>
                          <th>Fr</th>
                          <th>Sa</th>
                        </tr>
                      </thead>
                      <tbody>
                        <tr>
                          <td title="1999-12-26" class="ant-picker-cell">
                            <div class="ant-picker-cell-inner">26</div>
                          </td>
                          <td title="1999-12-27" class="ant-picker-cell">
                            <div class="ant-picker-cell-inner">27</div>
                          </td>
                          <td title="1999-12-28" class="ant-picker-cell">
                            <div class="ant-picker-cell-inner">28</div>
                          </td>
                          <td title="1999-12-29" class="ant-picker-cell">
                            <div class="ant-picker-cell-inner">29</div>
                          </td>
                          <td title="1999-12-30" class="ant-picker-cell">
                            <div class="ant-picker-cell-inner">30</div>
                          </td>
                          <td title="1999-12-31" class="ant-picker-cell ant-picker-cell-end">
                            <div class="ant-picker-cell-inner">31</div>
                          </td>
                          <td title="2000-01-01" class="ant-picker-cell ant-picker-cell-start ant-picker-cell-in-view ant-picker-cell-range-start ant-picker-cell-range-end ant-picker-cell-selected">
                            <div class="ant-picker-cell-inner">1</div>
                          </td>
                        </tr>
                        <tr>
                          <td title="2000-01-02" class="ant-picker-cell ant-picker-cell-in-view">
                            <div class="ant-picker-cell-inner">2</div>
                          </td>
                          <td title="2000-01-03" class="ant-picker-cell ant-picker-cell-in-view">
                            <div class="ant-picker-cell-inner">3</div>
                          </td>
                          <td title="2000-01-04" class="ant-picker-cell ant-picker-cell-in-view">
                            <div class="ant-picker-cell-inner">4</div>
                          </td>
                          <td title="2000-01-05" class="ant-picker-cell ant-picker-cell-in-view">
                            <div class="ant-picker-cell-inner">5</div>
                          </td>
                          <td title="2000-01-06" class="ant-picker-cell ant-picker-cell-in-view">
                            <div class="ant-picker-cell-inner">6</div>
                          </td>
                          <td title="2000-01-07" class="ant-picker-cell ant-picker-cell-in-view">
                            <div class="ant-picker-cell-inner">7</div>
                          </td>
                          <td title="2000-01-08" class="ant-picker-cell ant-picker-cell-in-view">
                            <div class="ant-picker-cell-inner">8</div>
                          </td>
                        </tr>
                        <tr>
                          <td title="2000-01-09" class="ant-picker-cell ant-picker-cell-in-view">
                            <div class="ant-picker-cell-inner">9</div>
                          </td>
                          <td title="2000-01-10" class="ant-picker-cell ant-picker-cell-in-view">
                            <div class="ant-picker-cell-inner">10</div>
                          </td>
                          <td title="2000-01-11" class="ant-picker-cell ant-picker-cell-in-view">
                            <div class="ant-picker-cell-inner">11</div>
                          </td>
                          <td title="2000-01-12" class="ant-picker-cell ant-picker-cell-in-view">
                            <div class="ant-picker-cell-inner">12</div>
                          </td>
                          <td title="2000-01-13" class="ant-picker-cell ant-picker-cell-in-view">
                            <div class="ant-picker-cell-inner">13</div>
                          </td>
                          <td title="2000-01-14" class="ant-picker-cell ant-picker-cell-in-view">
                            <div class="ant-picker-cell-inner">14</div>
                          </td>
                          <td title="2000-01-15" class="ant-picker-cell ant-picker-cell-in-view">
                            <div class="ant-picker-cell-inner">15</div>
                          </td>
                        </tr>
                        <tr>
                          <td title="2000-01-16" class="ant-picker-cell ant-picker-cell-in-view">
                            <div class="ant-picker-cell-inner">16</div>
                          </td>
                          <td title="2000-01-17" class="ant-picker-cell ant-picker-cell-in-view">
                            <div class="ant-picker-cell-inner">17</div>
                          </td>
                          <td title="2000-01-18" class="ant-picker-cell ant-picker-cell-in-view">
                            <div class="ant-picker-cell-inner">18</div>
                          </td>
                          <td title="2000-01-19" class="ant-picker-cell ant-picker-cell-in-view">
                            <div class="ant-picker-cell-inner">19</div>
                          </td>
                          <td title="2000-01-20" class="ant-picker-cell ant-picker-cell-in-view">
                            <div class="ant-picker-cell-inner">20</div>
                          </td>
                          <td title="2000-01-21" class="ant-picker-cell ant-picker-cell-in-view">
                            <div class="ant-picker-cell-inner">21</div>
                          </td>
                          <td title="2000-01-22" class="ant-picker-cell ant-picker-cell-in-view">
                            <div class="ant-picker-cell-inner">22</div>
                          </td>
                        </tr>
                        <tr>
                          <td title="2000-01-23" class="ant-picker-cell ant-picker-cell-in-view">
                            <div class="ant-picker-cell-inner">23</div>
                          </td>
                          <td title="2000-01-24" class="ant-picker-cell ant-picker-cell-in-view">
                            <div class="ant-picker-cell-inner">24</div>
                          </td>
                          <td title="2000-01-25" class="ant-picker-cell ant-picker-cell-in-view">
                            <div class="ant-picker-cell-inner">25</div>
                          </td>
                          <td title="2000-01-26" class="ant-picker-cell ant-picker-cell-in-view">
                            <div class="ant-picker-cell-inner">26</div>
                          </td>
                          <td title="2000-01-27" class="ant-picker-cell ant-picker-cell-in-view">
                            <div class="ant-picker-cell-inner">27</div>
                          </td>
                          <td title="2000-01-28" class="ant-picker-cell ant-picker-cell-in-view">
                            <div class="ant-picker-cell-inner">28</div>
                          </td>
                          <td title="2000-01-29" class="ant-picker-cell ant-picker-cell-in-view">
                            <div class="ant-picker-cell-inner">29</div>
                          </td>
                        </tr>
                        <tr>
                          <td title="2000-01-30" class="ant-picker-cell ant-picker-cell-in-view">
                            <div class="ant-picker-cell-inner">30</div>
                          </td>
                          <td title="2000-01-31" class="ant-picker-cell ant-picker-cell-in-view">
                            <div class="ant-picker-cell-inner">31</div>
                          </td>
                          <td title="2000-02-01" class="ant-picker-cell ant-picker-cell-start">
                            <div class="ant-picker-cell-inner">1</div>
                          </td>
                          <td title="2000-02-02" class="ant-picker-cell">
                            <div class="ant-picker-cell-inner">2</div>
                          </td>
                          <td title="2000-02-03" class="ant-picker-cell">
                            <div class="ant-picker-cell-inner">3</div>
                          </td>
                          <td title="2000-02-04" class="ant-picker-cell">
                            <div class="ant-picker-cell-inner">4</div>
                          </td>
                          <td title="2000-02-05" class="ant-picker-cell">
                            <div class="ant-picker-cell-inner">5</div>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
                <div class="ant-picker-time-panel">
                  <div class="ant-picker-header">
                    <!---->
                    <!---->
                    <div class="ant-picker-header-view">2000/01/01</div>
                    <!---->
                    <!---->
                  </div>
                  <div class="ant-picker-content"></div>
                </div>
              </div>
              <!---->
            </div>
          </div>
          <div class="ant-picker-footer">
            <!---->
            <ul class="ant-picker-ranges">
              <!---->
              <li class="ant-picker-ok"><button class="ant-btn ant-btn-primary ant-btn-sm" type="button">
                  <!----><span>Ok</span>
                </button></li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="ant-picker ant-picker-range">
    <div class="ant-picker-input ant-picker-input-active"><input readonly="" placeholder="Start date" size="12" autocomplete="off"></div>
    <div class="ant-picker-range-separator"><span aria-label="to" class="ant-picker-separator"><span role="img" aria-label="swap-right" class="anticon anticon-swap-right"><svg focusable="false" class="" data-icon="swap-right" width="1em" height="1em" fill="currentColor" aria-hidden="true" viewBox="0 0 1024 1024"><path d="M873.1 596.2l-164-208A32 32 0 00684 376h-64.8c-6.7 0-10.4 7.7-6.3 13l144.3 183H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h695.9c26.8 0 41.7-30.8 25.2-51.8z"></path></svg></span></span></div>
    <div class="ant-picker-input"><input readonly="" placeholder="End date" size="12" autocomplete="off"></div>
    <div class="ant-picker-active-bar" style="left: 0px; width: 0px; position: absolute;"></div><span class="ant-picker-suffix"><span role="img" aria-label="calendar" class="anticon anticon-calendar"><svg focusable="false" class="" data-icon="calendar" width="1em" height="1em" fill="currentColor" aria-hidden="true" viewBox="64 64 896 896"><path d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"></path></svg></span></span><span class="ant-picker-clear"><span role="img" aria-label="close-circle" class="anticon anticon-close-circle"><svg focusable="false" class="" data-icon="close-circle" width="1em" height="1em" fill="currentColor" aria-hidden="true" viewBox="64 64 896 896"><path d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"></path></svg></span></span>
  </div>
</div>
`;
